src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,54 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 1;
+    private final int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        int start = 0;
+        int workLoad = array.length;
+
+        if (workLoad > THRESHOLD) {
+            int middle = workLoad / 2;
+
+            int[] first = Arrays.copyOfRange(array, start, middle);
+            int[] second = Arrays.copyOfRange(array, middle, workLoad);
+
+            MergeSortAction firstSort = new MergeSortAction(first);
+            MergeSortAction secondSort = new MergeSortAction(second);
+
+            invokeAll(firstSort, secondSort);
+            merge(first, second);
+        }
+    }
+
+    private void merge(int[] first, int[] second) {
```

Comment:
The 'merge' method does not take into account the 'start' position of the subarray within the original array. As it stands, the merge will always start from the beginning of the original array. You need to modify the 'merge' method to handle merging at the correct positions within the original array.